### ISSUE TITLE
Question about the training script for the 0.02% (S3DIS)

### ISSUE THREAD

**Issue by jihun1998** [lizhaoliu-Lec/CPCM#4]
Hi.
Thank you for releasing the code.
However, I cannot find the training script for the 0.02% setting on the S3DIS dataset.
I modified the script for 0.01% as 0.02% and tried it.
Unfortunately, this setting fails to reproduce the reported performance.
Could you provide the official training script for it?
Any helpful comment will be greatly appreciated.

**Comment by lizhaoliu-Lec** [comment]
The hyper-parameters are almost the same for 0.01% and 0.02%. Which dataset are you using and what's the result you get?

**Comment by jihun1998** [comment]
Thank you for replying.
I am using s3dis dataset with configs for 0.01%.
I got about 42 mIoU in validation set (area 5) after 180 epoch training.

**Comment by lizhaoliu-Lec** [comment]
Maybe provide your training script and config file for me to further diagnosis?

**Comment by jihun1998** [comment]
I used same script which is provided from your repository.

CUDA_VISIBLE_DEVICES=0 python launch.py ddp_train.py --config config/default.yaml \
    GENERAL.exp_name 1e-4_percentage_consis_weight2_maskGrid075GridSize8_extraMaskStreamSelfCorr_weight10 \
    TRAINER.name TwoStreamTrainer \
    MODEL.out_channels 13 \
    DATA.name StanfordDataLoader \
    DATA.dataset StanfordArea5Dataset \
    DATA.voxel_size 0.05 \
    DATA.batch_size 2 \
    DATA.train_limit_numpoints 1000000 \
    OPTIMIZER.lr 0.01 \
    OPTIMIZER.weight_decay 0.001 \
    SCHEDULER.name PolyLR \
    TRAINER.epochs 180 \
    EVALUATOR.iou_num_class 13 \
    DATA.stanford3d_path /mnt/jihun3/s3dis_cpcm \
    DATA.stanford3d_sampled_inds /mnt/jihun3/CPCM/prepare_dataset/stanford/points/percentage0.0002evenc \
    DATA.sparse_label False \
    DATA.two_stream True \
    MODEL.two_stream_model_apply True \
    TRAINER.two_stream_feats_key semantic_scores \
    TRAINER.two_stream_loss_mode js_divergence_v2 \
    TRAINER.two_stream_seg_both True \
    TRAINER.two_stream_loss_weight 2.0 \
    AUGMENTATION.use_color_jitter False \
    TRAINER.two_stream_mask_grid_size 8 \
    TRAINER.two_stream_loss_mask_mode js_divergence_v2 \
    TRAINER.two_stream_mask_ratio 0.75 \
    TRAINER.two_stream_mask_mode grid \
    TRAINER.two_stream_mask_extra_stream True \
    TRAINER.two_stream_mask_feats_key semantic_scores \
    TRAINER.two_stream_mask_corr_loss True \
    TRAINER.two_stream_mask_self_loss True \
    TRAINER.two_stream_loss_mask_weight 10. \
    TRAINER.two_stream_mask_loss_threshold -1. \
    TRAINER.empty_cache_every 1

**Comment by lizhaoliu-Lec** [comment]
![image](https://github.com/lizhaoliu-Lec/CPCM/assets/38394970/51de3b6e-6888-4655-aa74-a0ea394debde)
I have run the experiments and my results are fine.
With consis weight = 2, mask weight = 10 (the script you provided), I get the mIoU of 61.16.
With consis weight = 5, mask weight = 5, I get the mIoU of 62.34.
The exp scripts are shared here: https://codefile.io/f/BP8u0JuRw9.

**Comment by jihun1998** [comment]
Thank you for your prompt response. :D

**Comment by xin1106** [comment]
> ![image](https://private-user-images.githubusercontent.com/38394970/278509730-51de3b6e-6888-4655-aa74-a0ea394debde.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxMDk1MDAsIm5iZiI6MTcxMzEwOTIwMCwicGF0aCI6Ii8zODM5NDk3MC8yNzg1MDk3MzAtNTFkZTNiNmUtNjg4OC00NjU1LWFhNzQtYTBlYTM5NGRlYmRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE0VDE1NDAwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2Yjk3Njg0MmI2YWI1YTc1YjBlNzYzODc4OGRlMWYxNTI5MzQ0M2MyZjU4MmUxMzRhZTU1ZDE0YmEwNDUyYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EoacIdETboT7KXYSdPxBPhkOIxtFcCOyX-WMfFt_04k) I have run the experiments and my results are fine. With consis weight = 2, mask weight = 10 (the script you provided), I get the mIoU of 61.16. With consis weight = 5, mask weight = 5, I get the mIoU of 62.34. The exp scripts are shared here: https://codefile.io/f/BP8u0JuRw9.

This URL (https://codefile.io/f/BP8u0JuRw9) is no longer available.  Could you provide the  exp scripts again?  I'm truly grateful for your help.

**Comment by YuenaQiao** [comment]
Hi, could you share your GPU settings? I ran the same script but only achieved a 25% val_mIoU (exp_3). I also tried two other scripts, but the final val_mIoU results were similarly poor. Could there be something I’m doing wrong? I followed the MinkowskiEngine 0.5.4 instructions, but the only difference is I’m using CUDA 11.3 instead of 11.6. My GPU is an NVIDIA GeForce RTX 3090. Could the problem be related to data preprocessing? Because I encountered an error while preparing the S3DIS data.
<img width="786" alt="CPCM_实验结果_val" src="https://github.com/user-attachments/assets/edc232f6-4cbb-405a-a6fb-390aae18c07d">

Thank you in advance for your help!

> I used same script which is provided from your repository.
> 
> CUDA_VISIBLE_DEVICES=0 python launch.py ddp_train.py --config config/default.yaml GENERAL.exp_name 1e-4_percentage_consis_weight2_maskGrid075GridSize8_extraMaskStreamSelfCorr_weight10 TRAINER.name TwoStreamTrainer MODEL.out_channels 13 DATA.name StanfordDataLoader DATA.dataset StanfordArea5Dataset DATA.voxel_size 0.05 DATA.batch_size 2 DATA.train_limit_numpoints 1000000 OPTIMIZER.lr 0.01 OPTIMIZER.weight_decay 0.001 SCHEDULER.name PolyLR TRAINER.epochs 180 EVALUATOR.iou_num_class 13 DATA.stanford3d_path /mnt/jihun3/s3dis_cpcm DATA.stanford3d_sampled_inds /mnt/jihun3/CPCM/prepare_dataset/stanford/points/percentage0.0002evenc DATA.sparse_label False DATA.two_stream True MODEL.two_stream_model_apply True TRAINER.two_stream_feats_key semantic_scores TRAINER.two_stream_loss_mode js_divergence_v2 TRAINER.two_stream_seg_both True TRAINER.two_stream_loss_weight 2.0 AUGMENTATION.use_color_jitter False TRAINER.two_stream_mask_grid_size 8 TRAINER.two_stream_loss_mask_mode js_divergence_v2 TRAINER.two_stream_mask_ratio 0.75 TRAINER.two_stream_mask_mode grid TRAINER.two_stream_mask_extra_stream True TRAINER.two_stream_mask_feats_key semantic_scores TRAINER.two_stream_mask_corr_loss True TRAINER.two_stream_mask_self_loss True TRAINER.two_stream_loss_mask_weight 10. TRAINER.two_stream_mask_loss_threshold -1. TRAINER.empty_cache_every 1
